src/core/components/resolve.js

```diff
@@ -1,64 +1,82 @@
 'use strict'
 
-const promisify = require('promisify-es6')
 const isIpfs = require('is-ipfs')
-const setImmediate = require('async/setImmediate')
 const CID = require('cids')
+const nodeify = require('promise-nodeify')
 const { cidToString } = require('../../utils/cid')
 
-module.exports = (self) => {
-  return promisify(async (name, opts, cb) => {
-    if (typeof opts === 'function') {
-      cb = opts
-      opts = {}
-    }
+/**
+ * @typedef { import("../index") } IPFS
+ */
 
-    opts = opts || {}
+/**
+ * @typedef {Object} ResolveOptions
+ * @prop {string} cidBase - Multibase codec name the CID in the resolved path will be encoded with
+ * @prop {boolean} [recursive=true] - Resolve until the result is an IPFS name
+ *
+ */
 
+/** @typedef {(err: Error, path: string) => void} ResolveCallback */
+
+/**
+ * @callback ResolveWrapper - This wrapper adds support for callbacks and promises
+ * @param {string} name - Path to resolve
+ * @param {ResolveOptions} opts - Options for resolve
+ * @param {ResolveCallback} [cb] - Optional callback function
+ * @returns {Promise<string> | void} - When callback is provided nothing is returned
+ */
+
+/**
+ * IPFS Resolve factory
+ *
+ * @param {IPFS} ipfs
+ * @returns {ResolveWrapper}
+ */
+module.exports = (ipfs) => {
+  /**
+   * IPFS Resolve - Resolve the value of names to IPFS
```

Comment:
```suggestion
   * IPFS Resolve - Resolve names to IPFS keys
```

Reply:
right now this is just a copy paste from the docs for consistency